src/index.js

```diff
@@ -134,13 +143,11 @@ function getUniqueSelector( element, selectorTypes, attributesToIgnore, filters
     if ( isDataAttributeSelectorType || isAttributeSelectorType )
     {
       const attributeToQuery = isDataAttributeSelectorType ? selectorType : selectorType.replace(attrRegex, '$1')
-      const attributeValue = element.getAttribute(attributeToQuery)
-      const attributeFilter = filters[selectorType];
-
+      const attributeSelector = getAttributeSelector(element, attributeToQuery, filter)
       // if we found a selector via attribute
-      if ( attributeValue !== null && (!attributeFilter || attributeFilter(selectorType, attributeToQuery, attributeValue)) )
+      if ( attributeSelector )
       {
-        selector = getAttributeSelector( element, attributeToQuery );
+        selector = getAttributeSelector( element, attributeToQuery, filter );
```

Comment:
Looks like this could just be: 

```suggestion
        selector = attributeSelector
```

And cut down on another potential filter execution?

Reply:
Yes! Great catch!